ECMA2Yaml/ECMAHelper/Models/ECMAStore.cs

```diff
@@ -1398,83 +1423,133 @@ private void SetInheritDocForType(Type t)
                 }
             }
 
-            DoValidation(t.Docs, t.Uid, t.SourceFileLocalPath);
+            DoValidation(t, t, true);
         }
 
         private void SetInheritDocForMember(Member m, Type t)
         {
             if (m.Signatures.IsStatic)
             {
-                OPSLogger.LogUserWarning(LogCode.ECMA2Yaml_Inheritdoc_InvalidTagsForStatic, t.SourceFileLocalPath, m.Uid);
+                OPSLogger.LogUserSuggestion(LogCode.ECMA2Yaml_Inheritdoc_InvalidTagsForStatic, t.SourceFileLocalPath, m.Uid);
                 return;
             }
 
             if (!IsNeedInheritdoc(t))
             {
                 return;
             }
-            if (!PreDoValidation(m.Docs, m.Uid, m.SourceFileLocalPath)) return;
 
-            if (m.ItemType == ItemType.Method || m.ItemType == ItemType.Constructor)
+            if (m.ItemType == ItemType.Method || m.ItemType == ItemType.Constructor || m.ItemType == ItemType.Property)
             {
-                var inheritDocId = m.Id;
-                if (t.InheritedMembersById?.Count > 0 && t.InheritedMembersById.ContainsKey(inheritDocId))
+                // 1. Get inheritdoc from cref object
+                if (!string.IsNullOrEmpty((m.Docs?.Inheritdoc.Cref)))
                 {
-                    var uids = t.InheritedMembersById[inheritDocId];
-                    if (uids.Count > 0)
+                    var crefUid = m.Docs?.Inheritdoc.Cref;
+                    if (MembersByUid.ContainsKey(crefUid))
                     {
-                        foreach (var uid in uids)
+                        var inheritFrom = MembersByUid[crefUid];
+                        if (SetInheritDoc(inheritFrom.Docs, m.Docs))
                         {
-                            if (MembersByUid.ContainsKey(uid))
+                            InheritDocItemsByUid.Add(m.Uid, inheritFrom.Uid);
+                        }
+                    }
+                    else
+                    {
+                        m.CrossInheritdocUid = crefUid;
+                        Console.WriteLine($"no found cref uid:{crefUid} for inheritdoc on uid:{m.Uid}-->");
```

Comment:
we don't need to write to console for this. We can comment out this line for now.

Reply:
updated.